htdocs/accountancy/journal/bankjournal.php

```diff
@@ -1004,141 +1008,86 @@ function getSourceDocRef($val, $typerecord)
 
 	// Defined the docref into $ref (We start with $val['ref'] by default and we complete according to other data)
 	// WE MUST HAVE SAME REF FOR ALL LINES WE WILL RECORD INTO THE BOOKKEEPING
-	$reflabel = $val['ref'];
-	if ($reflabel == '(SupplierInvoicePayment)' || $reflabel == '(SupplierInvoicePaymentBack)') {
-		$reflabel = $langs->trans('Supplier');
+	$ref = $val['ref'];
+	if ($ref == '(SupplierInvoicePayment)' || $ref == '(SupplierInvoicePaymentBack)') {
+		$ref = $langs->trans('Supplier');
 	}
-	if ($reflabel == '(CustomerInvoicePayment)' || $reflabel == '(CustomerInvoicePaymentBack)') {
-		$reflabel = $langs->trans('Customer');
+	if ($ref == '(CustomerInvoicePayment)' || $ref == '(CustomerInvoicePaymentBack)') {
+		$ref = $langs->trans('Customer');
 	}
-	if ($reflabel == '(SocialContributionPayment)') {
-		$reflabel = $langs->trans('SocialContribution');
+	if ($ref == '(SocialContributionPayment)') {
+		$ref = $langs->trans('SocialContribution');
 	}
-	if ($reflabel == '(DonationPayment)') {
-		$reflabel = $langs->trans('Donation');
+	if ($ref == '(DonationPayment)') {
+		$ref = $langs->trans('Donation');
 	}
-	if ($reflabel == '(SubscriptionPayment)') {
-		$reflabel = $langs->trans('Subscription');
+	if ($refl == '(SubscriptionPayment)') {
+		$ref = $langs->trans('Subscription');
 	}
-	if ($reflabel == '(ExpenseReportPayment)') {
-		$reflabel = $langs->trans('Employee');
+	if ($ref == '(ExpenseReportPayment)') {
+		$ref = $langs->trans('Employee');
 	}
-	if ($reflabel == '(payment_salary)') {
-		$reflabel = $langs->trans('Employee');
+	if ($ref == '(payment_salary)') {
+		$ref = $langs->trans('Employee');
 	}
-	$ref=$reflabel;
 	if ($typerecord == 'payment')
 	{
 		$sqlmid = 'SELECT payfac.fk_facture as id, f.facnumber as ref';
 		$sqlmid .= " FROM ".MAIN_DB_PREFIX."paiement_facture as payfac, ".MAIN_DB_PREFIX."facture as f";
 		$sqlmid .= " WHERE payfac.fk_facture = f.rowid AND payfac.fk_paiement=" . $val["paymentid"];
-		dol_syslog("accountancy/journal/bankjournal.php::sqlmid=" . $sqlmid, LOG_DEBUG);
-		$resultmid = $db->query($sqlmid);
-		if ($resultmid) {
-			$ref=$langs->trans("Invoice");
-			while ($objmid = $db->fetch_object($resultmid))
-			{
-				$ref.=' '.$objmid->ref;
-			}
-		}
-		else dol_print_error($db);
+        $ref = $langs->trans("Invoice");
```

Comment:
If one payment is done for several invoices, we are lossing here the list of all invoices into label. It seems a previous information. Why removing this ?